### PR TITLE
[OS-2728] middleware: adding to subscriptions

### PR DESCRIPTION
**Pull request by shruthiMaganahalli**
this is part 2 PR of this ticket https://blacklane.atlassian.net/browse/OS-2728

### CODE REVIEW

File: middleware/go.mod [hunk 1]
```diff
@@ -4,6 +4,7 @@ go 1.17
 
 require (
 	github.com/blacklane/go-libs/logger v0.6.0
+	github.com/blacklane/go-libs/camunda/v2 v2.0.0-20220208132721-238c65053cf6
```
Comment (shruthiMaganahalli) [inline middleware/go.mod:7]: i am looking into the weird version number

**Review by shane-blacklane — APPROVED** [review]
LGTM

File: middleware/subscriptions.go [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package middleware
+
+import (
+	"context"
+
+	"github.com/blacklane/go-libs/camunda/v2"
+	"github.com/blacklane/go-libs/logger"
+)
+
+func SubscriptionsAddLogger(log logger.Logger) func(camunda.TaskHandler) camunda.TaskHandler {
```
[the remark this answers is not in the document]
Comment (shruthiMaganahalli) [reply middleware/subscriptions.go]: done.

File: middleware/camundaSubscription_example_test.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package middleware_test
```
Comment (sabaraouf) [inline middleware/camundaSubscription_example_test.go:1]: please change the file name to snake case `camunda_subscription_example_test.go`

**Review by sabaraouf — COMMENTED** [review]
so I see that in the middleware folder there are 2 test case files, one is with `example` and the other without, do we need both?

also, can you update the readme https://github.com/blacklane/go-libs/blob/master/middleware/README.md

**Review by shane-blacklane — DISMISSED** [review]

File: middleware/camundaSubscription_example_test.go (hunk 3 above)
Comment (shruthiMaganahalli) [reply middleware/camundaSubscription_example_test.go]: oops! changed.

**Comment by sabaraouf** [comment]
> so I see that in the middleware folder there are 2 test case files, one is with `example` and the other without, do we need both?
> 
> also, can you update the readme https://github.com/blacklane/go-libs/blob/master/middleware/README.md

@shruthiMaganahalli ?

**Review by shane-blacklane — APPROVED** [review]

**Review by sabaraouf — APPROVED** [review]
🚀

**Review by giulliano-bueno — APPROVED** [review]

**Pull request merged**